src/bp/ui-admin/src/App/CommandPalette.tsx

```diff
@@ -0,0 +1,61 @@
+import { Commander, lang, QuickShortcut } from 'botpress/shared'
+import React, { useEffect, useState } from 'react'
+import { connect } from 'react-redux'
+import { withRouter } from 'react-router'
+
+import { fetchBots } from '../reducers/bots'
+
+const CommandPalette = props => {
+  const [commands, setCommands] = useState<QuickShortcut[]>([])
+
+  useEffect(() => {
+    if (!props.bots) {
+      props.fetchBots()
+    }
+
+    if (!props.workspaces || !props.bots) {
+      return
+    }
+
+    const getBotDisplayName = bot => {
+      return props.bots.filter(x => x.name === bot.name).length > 1 ? `${bot.name} (${bot.id})` : bot.name
+    }
+
+    const commands: QuickShortcut[] = []
+
+    for (const bot of props.bots) {
+      commands.push({
+        label: lang.tr('commander.viewBot', { name: getBotDisplayName(bot) }),
+        category: 'studio',
+        type: 'redirect',
+        url: window.location.origin + '/studio/' + bot.id
+      })
+    }
+
+    if (props.workspaces.length > 1) {
+      for (const workspace of props.workspaces) {
+        const [, , , urlPage] = props.location.pathname.split('/')
+        commands.push({
+          label: lang.tr('commander.switchWorkspace', { name: workspace.workspaceName }),
+          category: 'admin',
+          type: 'goto',
+          url: `/workspace/${workspace.workspaceName}/${urlPage}`
+        })
+      }
+    }
```

Comment:
Line 24 to 47 could be replaced by 
```const [, , , urlPage] = props.location.pathname.split('/')
setCommands([
  ...props.bots.map(bot => ({
    label: lang.tr('commander.viewBot', { name: getBotDisplayName(bot) }),
    category: 'studio',
    type: 'redirect',
    url: window.location.origin + '/studio/' + bot.id
  })),
  ...props.workspaces.map(workspace => ({
    label: lang.tr('commander.switchWorkspace', { name: workspace.workspaceName }),
    category: 'admin',
    type: 'goto',
    url: `/workspace/${workspace.workspaceName}/${urlPage}`
  }))
] as QuickShortcut[])
```

Reply:
Nop, because I don't want to add workspaces there if you only have one (most users don't have more)

Reply:
oh my bad I misread the condition, you can achieve it this way then ```props.workspaces.length > 1 ? props.workspaces.map(workspace => ({
    label: lang.tr('commander.switchWorkspace', { name: workspace.workspaceName }),
    category: 'admin',
    type: 'goto',
    url: `/workspace/${workspace.workspaceName}/${urlPage}`
  })) : []```

Reply:
Yes you are right, but I think i'll keep it that way, so we don't have a big clump of code for that, and it keeps it clean